contrib/dracut/systemd/zfs-dracut-bootfs.service.in

```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Set BOOTFS environment for dracut
+Documentation=man:zfs(8)
+DefaultDependencies=no
+After=zfs-import.target
+Requires=zfs-import.target
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -c "/bin/systemctl set-environment BOOTFS=$(@sbindir@/zpool list -H -o bootfs | grep -v '^-$' | head -1)"
```

Comment:
``head -1`` isn't part of the POSIX standard, ``head -n 1`` is

https://www.unix.com/man-page/posix/1P/HEAD/

Reply:
Well, I agree, but maybe we shouldn't mix too much in this PR. Making everything POSIX conformant may be worth another PR?

~~~
contrib/dracut/02zfsexpandknowledge/module-setup.sh.in:    numfields=`head -1 /proc/self/mountinfo | awk '{print NF}'`
config/kernel.m4:			             2>/dev/null | grep -v obj | tail -1)
config/deb.am:	arch=`$(RPM) -qp $${name}-kmod-$${version}.src.rpm --qf %{arch} | tail -1`; \
config/deb.am:	arch=`$(RPM) -qp $${name}-dkms-$${version}.src.rpm --qf %{arch} | tail -1`; \
config/deb.am:	arch=`$(RPM) -qp $${name}-$${version}.src.rpm --qf %{arch} | tail -1`; \
config/deb.am:	pkg10=`ls python*-pyzfs-$${version}* | tail -1`; \
config/tgz.am:	arch=`$(RPM) -qp $${name}-kmod-$${version}.src.rpm --qf %{arch} | tail -1`; \
config/tgz.am:	arch=`$(RPM) -qp $${name}-$${version}.src.rpm --qf %{arch} | tail -1`; \
debian/zfs-dkms.postinst:	addrlen=$(head -1 /proc/kallsyms| grep -o '^ *[^ ]*' |wc -c)
debian/zfs-dkms.config:	addrlen=$(head -1 /proc/kallsyms|awk '{print $1}'|wc -c)
tests/zfs-tests/tests/functional/cli_user/misc/arc_summary_001_pos.ksh:log_must eval "arc_summary | head -1 > /dev/null"
tests/zfs-tests/tests/functional/zvol/zvol_common.shlib:				tail -1 | awk '{print $3}')
tests/zfs-tests/tests/functional/history/history_003_pos.ksh:typeset orig_md5=$(zpool history $spool | head -2 | md5sum | \
tests/zfs-tests/tests/functional/history/history_003_pos.ksh:typeset final_md5=$(head -2 $TMPFILE | md5sum | awk '{print $1}')
tests/zfs-tests/tests/functional/bootfs/bootfs_006_pos.ksh:	VAL=$(zpool get bootfs $POOL | tail -1 | awk '{print $3}' )
tests/zfs-tests/tests/functional/bootfs/bootfs_006_pos.ksh:	VAL=$(zpool get bootfs $POOL | tail -1 | awk '{print $3}' )
tests/zfs-tests/tests/functional/bootfs/bootfs_003_pos.ksh:	RES=$(zpool get bootfs $POOL | tail -1 | awk '{print $3}' )
tests/zfs-tests/tests/functional/cli_root/zfs_create/zfs_create_012_pos.ksh:ZFS_VERSION=$(zfs upgrade | head -1 | awk '{print $NF}' \
tests/zfs-tests/tests/functional/cli_root/zpool_import/zpool_import.kshlib:		    awk '$3=="C" {print $1}' | tail -1)
tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh:		actual_clone=$(zfs list -t all -o clones $snapshot | tail -1)
tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_010_pos.ksh:char_count=$(echo "$clone_list" | tail -1 | wc | awk '{print $3}')
tests/zfs-tests/tests/functional/cli_root/zfs_clone/zfs_clone_007_pos.ksh:ZFS_VERSION=$(zfs upgrade | head -1 | awk '{print $NF}' \
tests/zfs-tests/tests/functional/cli_root/zfs_send/zfs_send_006_pos.ksh:		typeset total_size=$(zfs send $option $snapshot 2>&1 | tail -1)
tests/zfs-tests/tests/functional/cli_root/zfs_send/zfs_send_006_pos.ksh:		     2>&1 | tail -1)
tests/zfs-tests/tests/functional/cli_root/zpool_upgrade/zpool_upgrade_001_pos.ksh:log_must eval "zpool upgrade -v | head -1 | grep 'feature flags'"
tests/zfs-tests/tests/functional/slog/slog_015_neg.ksh:ORIG_TIMEOUT=$(get_tunable zfs_commit_timeout_pct | tail -1 | awk '{print $NF}')
tests/zfs-tests/include/libtest.shlib:			unit cyl print | head -3 | tail -1 | \
tests/zfs-tests/include/libtest.shlib:		prop_val=$(zpool get -pH $prop $pool 2>/dev/null | tail -1 | \
cmd/vdev_id/vdev_id:	PHY=`ls -d $port_dir/phy* 2>/dev/null | head -1 | awk -F: '{print $NF}'`
cmd/zed/zed.d/zed-functions.sh:        | tail -1 | cut -d\; -f1)"
scripts/zloop.sh:        printf "%s" "$(ls -tr1 $coreglob 2> /dev/null | head -1)"
~~~

Reply:
Sure, makes sense, I just happened by this while reading latest things in the repo.

Reply:
I agree, we don't want to make any other unrelated changes in this PR.  But let's use the standard POSIX version in `zfs-dracut-bootfs.service.in` while we're already making this change.

Aside from that this PR looks functionally correct to me, but we're going to want to test it with a dracut setup.